server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/DatasetService.java

```diff
@@ -371,4 +417,160 @@ public Map<String, String> signLinks(String project, String datasetName, Set<Str
             }
         }));
     }
+
+    @Transactional
+    public void build(CreateBuildRecordRequest request) {
+        var project = projectService.findProject(request.getProjectUrl());
+        if (request.getDatasetId() == null || request.getDatasetId() == 0) {
+            // create new dataset
+            // check the dataset name whether already in the same project
+            var ds = datasetMapper.findByName(request.getDatasetName(), project.getId(), true);
+            if (null != ds) {
+                throw new SwValidationException(ValidSubject.DATASET, MessageFormat.format(
+                    "The dataset:{0} in project:{1} is already exists, please use another name.",
+                    request.getDatasetName(), project.getName()));
+            }
+        } else {
+            var ds = datasetMapper.find(request.getDatasetId());
+            if (null == ds) {
+                throw new SwValidationException(ValidSubject.DATASET, MessageFormat.format(
+                    "The dataset:{0} in project:{1} doesn't exists, please check it.",
+                    request.getDatasetName(), project.getName()));
+            }
+            if (!ds.getDatasetName().equalsIgnoreCase(request.getDatasetName())) {
+                throw new SwValidationException(ValidSubject.DATASET, MessageFormat.format(
+                    "The dataset:{0} is different with the exist dataset:{1}.",
+                    request.getDatasetName(), ds.getDatasetName()));
+            }
+        }
+        var buildings = buildRecordMapper.selectBuildingInOneProjectForUpdate(
+                    project.getId(), request.getDatasetName());
```

Comment:
This line means we can apply two buildings with case-sensitive dataset names like `mnist` and `Mnist`?
And it seems that we do not allow this in Line 440:  
```java
if (!ds.getDatasetName().equalsIgnoreCase(request.getDatasetName())) {
```

Reply:
yes, you are right, i will fix it